spec/abstractops.html

```diff
@@ -548,51 +548,54 @@ <h1>
+      1. NOTE: For each singular Temporal unit name that is contained within _allowedStrings_, the corresponding plural name is also contained within it.
       1. If _default_ is ~unset~, then
         1. Let _defaultValue_ be *undefined*.
-      1. Else if _default_ is ~required~, then
-        1. Let _defaultValue_ be ~required~.
-      1. Else if _default_ is ~auto~, then
-        1. Append _default_ to _allowedValues_.
-        1. Let _defaultValue_ be *"auto"*.
       1. Else,
-        1. Assert: _allowedValues_ contains _default_.
-        1. Let _defaultValue_ be the value in the "Singular property name" column of <emu-xref href="#table-temporal-units"></emu-xref> corresponding to the row with _default_ in the "Value" column.
-      1. Let _allowedStrings_ be a new empty List.
-      1. For each element _value_ of _allowedValues_, do
-        1. If _value_ is ~auto~, then
-          1. Append *"auto"* to _allowedStrings_.
-        1. Else,
-          1. Let _singularName_ be the value in the "Singular property name" column of <emu-xref href="#table-temporal-units"></emu-xref> corresponding to the row with _value_ in the "Value" column.
-          1. Append _singularName_ to _allowedStrings_.
-          1. Let _pluralName_ be the value in the "Plural property name" column of the corresponding row.
-          1. Append _pluralName_ to _allowedStrings_.
-      1. NOTE: For each singular Temporal unit name that is contained within _allowedStrings_, the corresponding plural name is also contained within it.
+        1. Let _defaultValue_ be _default_.
       1. Let _value_ be ? GetOption(_options_, _key_, ~string~, _allowedStrings_, _defaultValue_).
       1. If _value_ is *undefined*, return ~unset~.
       1. If _value_ is *"auto"*, return ~auto~.
       1. Return the value in the "Value" column of <emu-xref href="#table-temporal-units"></emu-xref> corresponding to the row with _value_ in its "Singular property name" or "Plural property name" column.
     </emu-alg>
   </emu-clause>
 
+  <emu-clause id="sec-temporal-validatetemporalunitvaluedoption" type="abstract operation">
+    <h1>
+      ValidateTemporalUnitValue (
+        _value_: a Temporal unit, ~unset~, or ~auto~,
+        _unitGroup_: ~date~, ~time~, or ~datetime~,
+        optional _extraValues_: a List of either Temporal units or ~auto~,
+      ): either a normal completion containing ~unused~ or a throw completion
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>It validates that the result of GetTemporalUnitValuedOption is <emu-not-ref>covered</emu-not-ref> by the union of _unitGroup_ and _extraValues_.</dd>
+    </dl>
+    <emu-alg>
+      1. If _extraValues_ is present and _extraValues_ contains _value_, return ~unused~.
+      1. Let _category_ be the value in the “Category” column of the row of <emu-xref href="#table-temporal-units"></emu-xref> whose “Value” column contains _value_. If there is no such row, throw a RangeError exception.
```

Comment:
Markup nit:
```suggestion
      1. Let _category_ be the value in the “Category” column of the row of <emu-xref href="#table-temporal-units"></emu-xref> whose “Value” column contains _value_. If there is no such row, throw a *RangeError* exception.
```

Reply:
Also this throws if _value_ is UNSET, unless _extraValues_ explicitly contains UNSET as well. I'm not sure that is intentional... (see next comment)